### PR TITLE
Release 1.4.0

### PR DESCRIPTION
**Pull request by tcojean**
This is the release preparation branch. For now, mostly to run the pipeline and check everything is OK

### CODE REVIEW

**Review by upsj — APPROVED** [review]

File: third_party/gtest/CMakeLists.txt [hunk 1]
```diff
@@ -3,7 +3,7 @@ include(FetchContent)
 FetchContent_Declare(
     googletest
     GIT_REPOSITORY https://github.com/google/googletest.git
-    GIT_TAG        6a7ed316a5cdc07b6d26362c90770787513822d4
+    GIT_TAG        e2239ee6043f73722e7aa812a459f54a28552929
```
Comment (Slaedr) [inline third_party/gtest/CMakeLists.txt:6]: We don't want to use the latest version for this?

File: CHANGELOG.md [hunk 2]
```diff
@@ -6,6 +6,139 @@ commits. For a comprehensive list, use the following command:
...
+  [#675](https://github.com/ginkgo-project/ginkgo/pull/675), [#828](https://github.com/ginkgo-project/ginkgo/pull/828)
++ Add a Github bot system to facilitate development and PR management.
+  [#667](https://github.com/ginkgo-project/ginkgo/pull/667), [#674](https://github.com/ginkgo-project/ginkgo/pull/674), [#689](https://github.com/ginkgo-project/ginkgo/pull/689), [#853](https://github.com/ginkgo-project/ginkgo/pull/853)
++ Add Intel (DPC++) CI support and enable CI on HPC systems. [#736](https://github.com/ginkgo-project/ginkgo/pull/736), [#751](https://github.com/ginkgo-project/ginkgo/pull/751), [#781](https://github.com/ginkgo-project/ginkgo/pull/781)
++ Add ssh debugging for Github Actions CI. [#749](https://github.com/ginkgo-project/ginkgo/pull/749)
++ Add pipeline segmentation for better CI speed. [#737](https://github.com/ginkgo-project/ginkgo/pull/737)
+
+
+Changes:
++ Extends ISAI from triangular to general and spd matrices. [#690](https://github.com/ginkgo-project/ginkgo/pull/690)
```
Comment (Slaedr) [inline CHANGELOG.md:92]: Perhaps this should also be a feature and not just a change?

File: CHANGELOG.md [hunk 3]
```diff
@@ -6,6 +6,139 @@ commits. For a comprehensive list, use the following command:
...
++ Add benchmarks for BLAS operations [#772](https://github.com/ginkgo-project/ginkgo/pull/772), [#829](https://github.com/ginkgo-project/ginkgo/pull/829)
++ Add support for different precisions and consistent index types in benchmarks.
+  [#675](https://github.com/ginkgo-project/ginkgo/pull/675), [#828](https://github.com/ginkgo-project/ginkgo/pull/828)
++ Add a Github bot system to facilitate development and PR management.
+  [#667](https://github.com/ginkgo-project/ginkgo/pull/667), [#674](https://github.com/ginkgo-project/ginkgo/pull/674), [#689](https://github.com/ginkgo-project/ginkgo/pull/689), [#853](https://github.com/ginkgo-project/ginkgo/pull/853)
++ Add Intel (DPC++) CI support and enable CI on HPC systems. [#736](https://github.com/ginkgo-project/ginkgo/pull/736), [#751](https://github.com/ginkgo-project/ginkgo/pull/751), [#781](https://github.com/ginkgo-project/ginkgo/pull/781)
++ Add ssh debugging for Github Actions CI. [#749](https://github.com/ginkgo-project/ginkgo/pull/749)
++ Add pipeline segmentation for better CI speed. [#737](https://github.com/ginkgo-project/ginkgo/pull/737)
```
Comment (Slaedr) [inline CHANGELOG.md:88]:
```suggestion
+ Add pipeline segmentation for better CI speed. [#737](https://github.com/ginkgo-project/ginkgo/pull/737)
+ Row-major accessor is generalized to more than 2 dimensions and a new "block column-major" accessor has been added. [#707](https://github.com/ginkgo-project/ginkgo/pull/707)
```

**Review by Slaedr — APPROVED** [review]
Looks good, though I have some comments below.

File: CHANGELOG.md [hunk 4]
```diff
@@ -6,6 +6,139 @@ commits. For a comprehensive list, use the following command:
 git log --first-parent
 ```
 
+## Version 1.4.0
+
+The Ginkgo team is proud to announce the new Ginkgo minor release 1.4.0. This
+release brings most of the Ginkgo functionality to the Intel DPC++ ecosystem
+which enables Intel-GPU execution. The only Ginkgo features which have not been
+ported yet are some preconditioners. Ginkgo's mixed-precision support is greatly
+enhanced thanks to:
```
Comment (yhmtsai) [inline CHANGELOG.md:15]: maybe a linebreak before mixed-precision

**Review by yhmtsai — APPROVED** [review]

File: README.md [hunk 5]
```diff
@@ -62,6 +62,7 @@ The Ginkgo DPC++ module has the following __additional__ requirements:
 
 * _OneAPI 2021.3+_
 * Set `dpcpp` as the `CMAKE_CXX_COMPILER`
+* `C++17` is used to compile this module, while the rest of Ginkgo is compiled using `c++14`.
```
Comment (pratikvn) [inline README.md:65]: nit: I think this should either be `c++17` with a lower case `c`, or `C++ 17` with a space.

**Review by pratikvn — APPROVED** [review]
LGTM!

File: CHANGELOG.md [hunk 6]
```diff
@@ -6,6 +6,139 @@ commits. For a comprehensive list, use the following command:
...
+  [#667](https://github.com/ginkgo-project/ginkgo/pull/667), [#674](https://github.com/ginkgo-project/ginkgo/pull/674), [#689](https://github.com/ginkgo-project/ginkgo/pull/689), [#853](https://github.com/ginkgo-project/ginkgo/pull/853)
++ Add Intel (DPC++) CI support and enable CI on HPC systems. [#736](https://github.com/ginkgo-project/ginkgo/pull/736), [#751](https://github.com/ginkgo-project/ginkgo/pull/751), [#781](https://github.com/ginkgo-project/ginkgo/pull/781)
++ Add ssh debugging for Github Actions CI. [#749](https://github.com/ginkgo-project/ginkgo/pull/749)
++ Add pipeline segmentation for better CI speed. [#737](https://github.com/ginkgo-project/ginkgo/pull/737)
+
+
+Changes:
++ Extends ISAI from triangular to general and spd matrices. [#690](https://github.com/ginkgo-project/ginkgo/pull/690)
++ Add a Scalar Jacobi specialization and kernels. [#808](https://github.com/ginkgo-project/ginkgo/pull/808), [#834](https://github.com/ginkgo-project/ginkgo/pull/834), [#854](https://github.com/ginkgo-project/ginkgo/pull/854)
++ Add implicit residual log for solvers and benchmarks. [#707](https://github.com/ginkgo-project/ginkgo/pull/707)
```
Comment (yhmtsai) [inline CHANGELOG.md:94]:
```suggestion
+ Add implicit residual log for solvers and benchmarks. [#714](https://github.com/ginkgo-project/ginkgo/pull/714)
```
is it 714 not 707?

File: CHANGELOG.md [hunk 7]
```diff
@@ -6,6 +6,139 @@ commits. For a comprehensive list, use the following command:
...
++ Add the IDR(s) solver. [#620](https://github.com/ginkgo-project/ginkgo/pull/620)
++ Add a new fixed-size block CSR matrix format (for the Reference executor).
+  [#671](https://github.com/ginkgo-project/ginkgo/pull/671), [#730](https://github.com/ginkgo-project/ginkgo/pull/730)
++ Add native mixed-precision support to the ELL format. [#717](https://github.com/ginkgo-project/ginkgo/pull/717), [#780](https://github.com/ginkgo-project/ginkgo/pull/780)
++ Add Reverse Cuthill-McKee reordering [#500](https://github.com/ginkgo-project/ginkgo/pull/500), [#649](https://github.com/ginkgo-project/ginkgo/pull/649)
++ Add matrix assembly support on CPUs. [#644](https://github.com/ginkgo-project/ginkgo/pull/644)
+
+Other additions:
++ Add possibility to apply real matrices to complex vectors.
+  [#655](https://github.com/ginkgo-project/ginkgo/pull/655), [#636](https://github.com/ginkgo-project/ginkgo/pull/636), [#658](https://github.com/ginkgo-project/ginkgo/pull/658)
```
Comment (yhmtsai) [inline CHANGELOG.md:72]: Is 636 wrong?

**Comment by hartwiganzt** [comment]
Congrats!

**Pull request merged**